plaid-stl/src/npm/mod.rs

```diff
@@ -0,0 +1,607 @@
+    Public,
+}
+
+impl PkgAccessLevel {
+    pub fn to_string(&self) -> String {
+        match self {
+            PkgAccessLevel::Restricted => "restricted".to_string(),
+            PkgAccessLevel::Public => "public".to_string(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct PublishEmptyStubParams {
+    pub package_name: String,
+    pub access_level: PkgAccessLevel,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ListPackagesWithTeamPermissionParams {
+    pub team: NpmTeam,
+    pub permission: NpmPackagePermission,
+}
+
+// End of "All the structs which are also used in the Plaid API"
+
+/// Retrieve a list of users in the npm organization
+pub fn get_org_user_list() -> Result<Vec<NpmUser>, PlaidFunctionError> {
+    extern "C" {
+        new_host_function_with_error_buffer!(npm, get_org_user_list);
+    }
+    const RETURN_BUFFER_SIZE: usize = 32 * 1024; // 32 KiB
+    let mut return_buffer = vec![0; RETURN_BUFFER_SIZE];
+
+    let params = "".to_string();
+
+    let res = unsafe {
+        npm_get_org_user_list(
+            params.as_bytes().as_ptr(),
+            params.as_bytes().len(),
+            return_buffer.as_mut_ptr(),
+            RETURN_BUFFER_SIZE,
+        )
+    };
+
+    if res < 0 {
+        return Err(res.into());
+    }
+
+    return_buffer.truncate(res as usize);
+    // This should be safe because unless the Plaid runtime is expressly trying
+    // to mess with us, this came from a String in the API module.
+    let res = String::from_utf8(return_buffer).unwrap();
+
+    serde_json::from_str::<Vec<NpmUser>>(&res).map_err(|_| PlaidFunctionError::InternalApiError)
+}
+
+/// Retrieve a list of users in the npm organization that do not have 2FA enabled
+pub fn get_org_users_without_2fa() -> Result<Vec<NpmUser>, PlaidFunctionError> {
+    extern "C" {
+        new_host_function_with_error_buffer!(npm, get_org_users_without_2fa);
+    }
+    const RETURN_BUFFER_SIZE: usize = 32 * 1024; // 32 KiB
+    let mut return_buffer = vec![0; RETURN_BUFFER_SIZE];
+
+    let params = "".to_string();
+
+    let res = unsafe {
+        npm_get_org_users_without_2fa(
+            params.as_bytes().as_ptr(),
+            params.as_bytes().len(),
+            return_buffer.as_mut_ptr(),
+            RETURN_BUFFER_SIZE,
+        )
+    };
+
+    if res < 0 {
+        return Err(res.into());
+    }
+
+    return_buffer.truncate(res as usize);
+    // This should be safe because unless the Plaid runtime is expressly trying
+    // to mess with us, this came from a String in the API module.
+    let res = String::from_utf8(return_buffer).unwrap();
+
+    serde_json::from_str::<Vec<NpmUser>>(&res).map_err(|_| PlaidFunctionError::InternalApiError)
+}
+
+/// Invite a user to join the npm organization. If the user accepts the invite, they will be added
+/// to the default team "developers".
+///
+/// TODO The Plaid API supports specifying another team as the default for the new user. Expose through the STL?
```

Comment:
TODO in code.

Not sure how to understand the question though.

Reply:
When inviting a user into an organization, the default is to add the user to the `developers` team. However, npm (and the API) support inviting a user _directly_ into another team, like `Admins`, instead of inviting and then promoting.
We can expose this through the STL and take it as a parameter, if we think there is no risk of abuse or error from the rule's part.